### PR TITLE
coretasks: add tests

### PR DESCRIPTION
**Pull request by HumorBaby**
Test coverage for `coretasks.py` is a little lacking. Here is my attempt to remedy that... :hankey: 

Requires #1542 to be merged first.

### CODE REVIEW

**Comment by HumorBaby** [comment]
Inspired by #1536 and my desire to delay all other responsibilities I have :smile_cat:.

Since I took the first test commit  from #1536 (`test: layout tests for sopel/coretasks.py`), there will definitly be merge conflicts later :fire:

**Comment by dgw** [comment]
When this is out of draft status and ready for review, request it from Exirel, since he's pretty much "the tests guy" at this point.

File: test/test_coretasks.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        return bot
+    return _fn
+
+
+def test_pass(dummy_bot_trigger):
+    bot, trigger = dummy_bot_trigger
+    assert True
+
+
+def test_auth_nickserv(bot_with_auth, mocker):
```
Comment (HumorBaby) [inline test/test_coretasks.py]: Need to add `pytest-mock` to `dev-requirements.txt`.

**Comment by Exirel** [comment]
@HumorBaby as I said in another of your PR, your tests are a bit too convoluted. In particular, your fixtures are hard to follow, some of them returning function, etc. it should be easier to read your tests and we should not have to read 30 lines of fixtures to add or change one.

**Comment by Exirel** [comment]
Oh, and we now have tests on coretasks, so you'll need to rebase your work from master if you want to proceed further.

**Comment by dgw** [comment]
@HumorBaby Are you planning to continue this at some point? I'm going to remove it from 7.0 milestone for now, since we already do have at least _some_ coretasks tests ready and this needs a fair amount of work before it could be ready to merge.

**Comment by HumorBaby** [comment]
@Exirel :
> Oh, and we now have tests on coretasks, ...

@dgw:
> since we already do have at least some coretasks tests ready...

My intention with this PR was to simply _add_ tests (which were completely missing at the time) for `coretasks`; the main tests that I was planning/had built (poorly) were for #1536, which has now been rebased and uses the existing `test_coretasks.py`, thus obviating this PR. I did not have anything more specific in mind and was planning on just building the `test_coretasks.py` file as I read/through learned the inner-workings of `coretasks`. 

So, since we now have tests for `coretasks`<sup>(:tada:)</sup>, I am closing this PR in favor of newer PRs that add/modify `coretasks` functionality appending/modifying the tests in the lovely `test_coretasks.py` that already exists.

**Comment by dgw** [comment]
Found a new use for our "Replaced" label, haha. Rather than making a new "Obsolete" one. 😁

Sounds all good to me, @HumorBaby. To the future we go!
